### PR TITLE
[Feature] Support RoITransRoIHead

### PR DESCRIPTION
**Pull request by nijkah**
## Motivation
Support `RoITransRoIHead` for RoI-Transformer in MMrotate.
**Note that it can be used after https://github.com/open-mmlab/mmdeploy/pull/650 is merged, which fixes `rotated_rpn_head`.**

## Modification

Add Rewriter for `RoITransRoIHead.simple_test`.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

- [x] add rewritings for RoITransRoIHead
- [x] Performance check with TensorRT model
- [x] Add docs
- [x] Add a unit test

<div style="margin-left: 25px;">
<table class="docutils">
<thead>
  <tr>
    <th align="center" colspan="4">mmrotate</th>
    <th align="center">Pytorch</th>
    <th align="center">ONNXRuntime</th>
    <th align="center" colspan="2">TensorRT</th>
    <th align="center">PPLNN</th>
    <th align="center">OpenVINO</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td align="center">model</td>
    <td align="center">task</td>
    <td align="center">dataset</td>
    <td align="center">metrics</td>
    <td align="center">fp32</td>
    <td align="center">fp32</td>
    <td align="center">fp32</td>
    <td align="center">fp16</td>
    <td align="center">fp16</td>
    <td align="center">fp32</td>
  </tr>
  <tr>
    <td align="center"><a href="https://github.com/open-mmlab/mmrotate/blob/main/configs/roi_trans/roi_trans_r50_fpn_1x_dota_le90.py">RoITransformer R50</a></td>
    <td align="center">Rotated Detection</td>
    <td align="center">DOTA-v1.0</td>
    <td align="center">mAP</td>
    <td align="center">0.761</td>
    <td align="center">-</td>
    <td align="center">0.758</td>
    <td align="center">-</td>
    <td align="center">-</td>
    <td align="center">-</td>
  </tr>
  <tr>
    <td align="center"><a href="https://github.com/open-mmlab/mmrotate/blob/main/configs/roi_trans/roi_trans_swin_tiny_fpn_1x_dota_le90.py">RoITransformer Swin-tiny</a></td>
    <td align="center">Rotated Detection</td>
    <td align="center">DOTA-v1.0</td>
    <td align="center">mAP</td>
    <td align="center">0.775</td>
    <td align="center">-</td>
    <td align="center">0.774</td>
    <td align="center">-</td>
    <td align="center">-</td>
    <td align="center">-</td>
  </tr>
</tbody>
</table>
</div>

### CODE REVIEW

File: mmdeploy/codebase/mmrotate/models/roi_trans_roi_head.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import torch
+
+from mmdeploy.core import FUNCTION_REWRITER
+
+
+@FUNCTION_REWRITER.register_rewriter(
+    'mmrotate.models.roi_heads.roi_trans_roi_head'
+    '.RoITransRoIHead.simple_test')
+def roi_trans_roi_head__simple_test(ctx, self, x, proposals, img_metas,
```
Comment (RunningLeon) [inline mmdeploy/codebase/mmrotate/models/roi_trans_roi_head.py:10]: We could add a Unit test for this rewriting in https://github.com/open-mmlab/mmdeploy/blob/master/tests/test_codebase/test_mmrotate/test_mmrotate_models.py

File: docs/en/03-benchmark/benchmark.md [hunk 2]
```diff
@@ -1641,6 +1641,19 @@ Users can directly test the performance through [how_to_evaluate_a_model.md](../
     <td align="center">-</td>
     <td align="center">-</td>
   </tr>
+  <tr>
+    <td align="center"><a href="https://github.com/open-mmlab/mmrotate/blob/main/configs/roi_trans/roi_trans_r50_fpn_1x_dota_le90.py">RoI Transformer</a></td>
```
Comment (RunningLeon) [inline docs/en/03-benchmark/benchmark.md:1645]: Hi, could you add the new model config in regression test yml for easy test?  https://github.com/open-mmlab/mmdeploy/blob/69c38b9f74be84084f4b8ee357d9ebfeefb3c69f/tests/regression/mmrotate.yml#L43

File: mmdeploy/codebase/mmrotate/models/roi_trans_roi_head.py (hunk 1 above)
Comment (nijkah) [reply mmdeploy/codebase/mmrotate/models/roi_trans_roi_head.py:10]: Can I ask for help in adding a unit test?
I pushed the draft for the unit test.
But it throws the error like below;
```
        wrapped_model = WrapModel(roi_head, 'simple_test', img_metas=img_metas)
>       rewrite_outputs, is_backend_output = get_rewrite_outputs(
            wrapped_model=wrapped_model,
            model_inputs=model_inputs,
            deploy_cfg=deploy_cfg)

tests/test_codebase/test_mmrotate/test_mmrotate_models.py:457:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
mmdeploy/utils/test.py:591: in get_rewrite_outputs
    backend_outputs = get_backend_outputs(ir_file_path, model_inputs,
mmdeploy/utils/test.py:553: in get_backend_outputs
    backend_outputs = backend_model(backend_feats)
/opt/conda/lib/python3.8/site-packages/torch/nn/modules/module.py:1102: in _call_impl
    return forward_call(*input, **kwargs)
mmdeploy/backend/onnxruntime/wrapper.py:85: in forward
    self.io_binding.bind_input(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <onnxruntime.capi.onnxruntime_inference_collection.IOBinding object at 0x7fb5a0ef9b20>, name = 'proposals_1', device_type = 'cpu', device_id = -1, element_type = dtype('float32'), shape = torch.Size([1, 1, 1])
buffer_ptr = 94579117147072

    def bind_input(self, name, device_type, device_id, element_type, shape, buffer_ptr):
        '''
        :param name: input name
        :param device_type: e.g. cpu, cuda
        :param device_id: device id, e.g. 0
        :param element_type: input element type
        :param shape: input shape
        :param buffer_ptr: memory pointer to input data
        '''
>       self._iobinding.bind_input(name,
                                   C.OrtDevice(get_ort_device_type(device_type), C.OrtDevice.default_memory(),
                                               device_id),
                                   element_type, shape, buffer_ptr)
E       RuntimeError: Error when binding input: Failed to find input name in the mapping: proposals_1

/opt/conda/lib/python3.8/site-packages/onnxruntime/capi/onnxruntime_inference_collection.py:381: RuntimeError
```

Maybe, the trouble part is feeding the input;
```
    model_inputs = {
        'x': x,
        'proposals': [proposals, labels],
    }

```

**Comment by nijkah** [comment]
I found that sometimes the predicted result is quite different compared to the pytorch model.
Tested with [roi_trans_swin_tiny_fpn_1x_dota_le90](https://github.com/open-mmlab/mmrotate/blob/main/configs/roi_trans/roi_trans_swin_tiny_fpn_1x_dota_le90.py)
```
# root @ ws-hakjin-6db678b6-nvvqr in ~/workspace/mmrotate on git:main x [5:44:50]
$ head ~/workspace/mmdeploy/roi_trans_2643/Task1_results/Task1_baseball-diamond.txt                                                                                                                                                    31s
P2643 0.637549102306366 4702.72 3994.82 4804.41 3979.51 4819.57 4080.20 4717.88 4095.52
P2643 0.5205038189888 4357.66 3713.31 4459.15 3715.00 4457.63 3806.84 4356.13 3805.15
P2643 0.2156124860048294 3813.04 1393.61 4096.71 1429.80 4067.03 1662.41 3783.37 1626.22
P2643 0.17779527604579926 4675.29 3776.19 4789.58 3777.87 4788.11 3877.54 4673.82 3875.86
P2643 0.13109613955020905 4043.61 1685.51 4193.22 1657.14 4220.51 1801.02 4070.89 1829.39
P2643 0.11649016290903091 4798.06 3864.99 4907.36 3865.03 4907.32 3969.31 4798.02 3969.26
P2643 0.07121634483337402 5127.67 1503.08 5233.58 1514.99 5221.62 1621.30 5115.71 1609.39
P2643 0.06801088899374008 4566.45 3689.70 4668.94 3691.34 4667.43 3785.29 4564.94 3783.65

# root @ ws-hakjin-6db678b6-nvvqr in ~/workspace/mmrotate on git:main x [5:45:27]
$ head roi_trains_2643/Task1_results/Task1_baseball-diamond.txt                                                                                                                                                                         0s
P2643 0.698990523815155 4709.87 3986.23 4813.27 3987.07 4812.44 4089.49 4709.03 4088.65
P2643 0.5733247995376587 4366.51 3707.61 4462.31 3721.79 4448.91 3812.37 4353.11 3798.19
P2643 0.21527588367462158 3813.04 1393.60 4096.72 1429.80 4067.03 1662.42 3783.36 1626.22
P2643 0.2114844024181366 4564.14 3693.04 4667.32 3690.62 4669.47 3782.47 4566.29 3784.88
P2643 0.1629500836133957 4042.65 1700.25 4193.81 1666.58 4221.50 1790.89 4070.35 1824.57
P2643 0.12476478517055511 4670.28 3782.63 4790.91 3778.71 4793.93 3871.49 4673.29 3875.41
P2643 0.11624093353748322 4790.71 3878.29 4899.42 3859.01 4917.20 3959.27 4808.49 3978.55
P2643 0.08829200267791748 5123.39 1503.52 5229.69 1507.54 5225.33 1622.82 5119.03 1618.80
```

**Comment by grimoire** [comment]
@nijkah I have fixed a bug of TRT Rotated NMS in #650. Not sure if it is related to the accuracy.

**Comment by RunningLeon** [comment]
@nijkah Hi, since https://github.com/open-mmlab/mmdeploy/pull/650 is merged, could merge master and fix the conflicts?

File: mmdeploy/codebase/mmrotate/models/roi_trans_roi_head.py (hunk 1 above)
Comment (RunningLeon) [reply mmdeploy/codebase/mmrotate/models/roi_trans_roi_head.py:10]: does proposals have `labels` in it?

File: mmdeploy/codebase/mmrotate/models/roi_heads/roi_trans_roi_head.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import torch
+
+from mmdeploy.core import FUNCTION_REWRITER
+
+
+@FUNCTION_REWRITER.register_rewriter(
+    'mmrotate.models.roi_heads.roi_trans_roi_head'
+    '.RoITransRoIHead.simple_test')
+def roi_trans_roi_head__simple_test(ctx, self, x, proposals, img_metas,
```
Comment (RunningLeon) [inline mmdeploy/codebase/mmrotate/models/roi_heads/roi_trans_roi_head.py:10]: we should keep the argument `proposals` same as the original `proposal_list` in https://github.com/open-mmlab/mmrotate/blob/c62f148fcc2c8253218e67f2277cb8770bcd7df0/mmrotate/models/roi_heads/roi_trans_roi_head.py#L266

File: tests/test_codebase/test_mmrotate/test_mmrotate_models.py [hunk 4]
```diff
@@ -552,3 +552,131 @@ def test_gv_ratio_roi_head__simple_test(backend_type: Backend):
...
+            codebase_config=dict(
+                type='mmrotate',
+                task='RotatedDetection',
+                post_processing=dict(
+                    score_threshold=0.05,
+                    iou_threshold=0.1,
+                    pre_top_k=2000,
+                    keep_top_k=2000))))
+
+    wrapped_model = WrapModel(roi_head, 'simple_test', img_metas=img_metas)
```
Comment (RunningLeon) [inline tests/test_codebase/test_mmrotate/test_mmrotate_models.py:676]: to make the unit test work, we could set a default value to `proposal_list` in here  as `proposal_list =[proposals, labels]` and remove it from `model_inputs`

**Comment by nijkah** [comment]
> I found that the predicted result is quite different compared to the pytorch model in some scenes. Tested with [roi_trans_swin_tiny_fpn_1x_dota_le90](https://github.com/open-mmlab/mmrotate/blob/main/configs/roi_trans/roi_trans_swin_tiny_fpn_1x_dota_le90.py)
> 
> ```
> # root @ ws-hakjin-6db678b6-nvvqr in ~/workspace/mmrotate on git:main x [5:44:50]
> $ head ~/workspace/mmdeploy/roi_trans_2643/Task1_results/Task1_baseball-diamond.txt                                                                                                                                                    31s
> P2643 0.637549102306366 4702.72 3994.82 4804.41 3979.51 4819.57 4080.20 4717.88 4095.52
> P2643 0.5205038189888 4357.66 3713.31 4459.15 3715.00 4457.63 3806.84 4356.13 3805.15
> P2643 0.2156124860048294 3813.04 1393.61 4096.71 1429.80 4067.03 1662.41 3783.37 1626.22
> P2643 0.17779527604579926 4675.29 3776.19 4789.58 3777.87 4788.11 3877.54 4673.82 3875.86
> P2643 0.13109613955020905 4043.61 1685.51 4193.22 1657.14 4220.51 1801.02 4070.89 1829.39
> P2643 0.11649016290903091 4798.06 3864.99 4907.36 3865.03 4907.32 3969.31 4798.02 3969.26
> P2643 0.07121634483337402 5127.67 1503.08 5233.58 1514.99 5221.62 1621.30 5115.71 1609.39
> P2643 0.06801088899374008 4566.45 3689.70 4668.94 3691.34 4667.43 3785.29 4564.94 3783.65
> 
> # root @ ws-hakjin-6db678b6-nvvqr in ~/workspace/mmrotate on git:main x [5:45:27]
> $ head roi_trans_2643/Task1_results/Task1_baseball-diamond.txt                                                                                                                                                                         0s
> P2643 0.698990523815155 4709.87 3986.23 4813.27 3987.07 4812.44 4089.49 4709.03 4088.65
> P2643 0.5733247995376587 4366.51 3707.61 4462.31 3721.79 4448.91 3812.37 4353.11 3798.19
> P2643 0.21527588367462158 3813.04 1393.60 4096.72 1429.80 4067.03 1662.42 3783.36 1626.22
> P2643 0.2114844024181366 4564.14 3693.04 4667.32 3690.62 4669.47 3782.47 4566.29 3784.88
> P2643 0.1629500836133957 4042.65 1700.25 4193.81 1666.58 4221.50 1790.89 4070.35 1824.57
> P2643 0.12476478517055511 4670.28 3782.63 4790.91 3778.71 4793.93 3871.49 4673.29 3875.41
> P2643 0.11624093353748322 4790.71 3878.29 4899.42 3859.01 4917.20 3959.27 4808.49 3978.55
> P2643 0.08829200267791748 5123.39 1503.52 5229.69 1507.54 5225.33 1622.82 5119.03 1618.80
> ```

@RunningLeon, I am debugging the performance gap between the pytorch model and ONNX & TRT models.
Until now, I found that [the result from the TRT wrapper](https://github.com/open-mmlab/mmdeploy/blob/71d085b1a69187744aac9447e96cf5dfdb47d6d6/mmdeploy/backend/tensorrt/wrapper.py#L167) and [the result from the pytorch model](https://github.com/open-mmlab/mmrotate/blob/c62f148fcc2c8253218e67f2277cb8770bcd7df0/mmrotate/models/detectors/two_stage.py#L183) are same exactly.
But the merged results are quite different like 
```
# root @ ws-hakjin-69b98d88f6-sm7sz in ~/workspace/mmdeploy on git:fa57490 x [9:23:39] C:130
$ head roi_trans_r50_new_trt/Task1_results/Task1_baseball-diamond.txt                                                                                                                                                                  58s
P2643 0.7715607285499573 3740.97 1459.14 4017.96 1389.74 4064.57 1575.78 3787.58 1645.17
P2643 0.733237624168396 4710.55 3990.44 4813.93 3984.44 4819.73 4084.44 4716.35 4090.44
P2643 0.5749783515930176 4351.68 3725.28 4454.60 3708.67 4469.04 3798.17 4366.12 3814.78
P2643 0.5543936491012573 4570.47 3689.56 4668.08 3688.10 4669.53 3784.78 4571.91 3786.24
P2643 0.5379559397697449 4473.94 3808.90 4575.48 3805.35 4578.92 3904.00 4477.39 3907.54
P2643 0.46451470255851746 4666.36 3789.71 4774.49 3766.58 4796.84 3871.09 4688.71 3894.22
P2643 0.3679143488407135 3994.14 1690.57 4183.64 1632.95 4237.47 1809.98 4047.96 1867.60
P2643 0.27642694115638733 5123.77 1526.21 5217.41 1516.35 5225.56 1593.80 5131.92 1603.65
P2643 0.2661984860897064 4447.88 3595.96 4552.39 3602.76 4546.04 3700.35 4441.53 3693.55
P2643 0.24539612233638763 4803.45 3866.50 4909.08 3874.35 4901.79 3972.31 4796.17 3964.45

# root @ ws-hakjin-69b98d88f6-sm7sz in ~/workspace/mmdeploy on git:fa57490 x [9:23:44]
$ head ../mmrotate/roi_trans_r50/Task1_results/Task1_baseball-diamond.txt                                                                                                                                                               0s
P2643 0.7718087434768677 3740.95 1459.16 4017.98 1389.70 4064.62 1575.74 3787.59 1645.20
P2643 0.7333663702011108 4710.55 3990.44 4813.93 3984.45 4819.72 4084.44 4716.34 4090.43
P2643 0.5752703547477722 4351.69 3725.28 4454.60 3708.68 4469.04 3798.18 4366.13 3814.78
P2643 0.554319441318512 4570.46 3689.57 4668.08 3688.09 4669.55 3784.76 4571.93 3786.25
P2643 0.5378356575965881 4473.93 3808.91 4575.46 3805.34 4578.94 3903.98 4477.40 3907.55
P2643 0.4634431302547455 4666.34 3789.74 4774.46 3766.56 4796.87 3871.07 4688.74 3894.25
P2643 0.431684672832489 5122.15 1530.39 5211.60 1512.09 5227.39 1589.27 5137.94 1607.57
P2643 0.40507709980010986 4025.09 1672.99 4196.97 1659.46 4210.63 1832.99 4038.75 1846.52
P2643 0.35206887125968933 4795.44 3873.99 4903.55 3864.99 4911.75 3963.35 4803.63 3972.35
P2643 0.26584571599960327 4442.42 3600.27 4545.86 3593.95 4551.90 3692.87 4448.46 3699.19
```

I wonder where the [postprocessing_params](https://github.com/open-mmlab/mmdeploy/blob/71d085b1a69187744aac9447e96cf5dfdb47d6d6/configs/mmrotate/rotated-detection_static.py#L5) are used in. I suspect the difference arises the postprocessing part.

File: mmdeploy/codebase/mmrotate/models/roi_heads/roi_trans_roi_head.py (hunk 3 above)
Comment (nijkah) [reply mmdeploy/codebase/mmrotate/models/roi_heads/roi_trans_roi_head.py:10]: I'll try it.

**Comment by RunningLeon** [comment]
@nijkah Hi, postprocess params are used in nms like here: https://github.com/open-mmlab/mmdeploy/blob/d216af885d556caca5ddf1f6b532e0395cac593c/mmdeploy/codebase/mmrotate/models/dense_heads/rotated_rpn_head.py#L129:143

**Comment by grimoire** [comment]
Hi, which ONNXRuntime version are you using? The unit test failed with log:

```
onnxruntime.capi.onnxruntime_pybind11_state.NotImplemented: [ONNXRuntimeError] : 9 : NOT_IMPLEMENTED : Could not find an implementation for Floor(6) node with name 'Floor_4'
```

**Comment by nijkah** [comment]
> Hi, which ONNXRuntime version are you using? The unit test failed with log:
> 
> ```
> onnxruntime.capi.onnxruntime_pybind11_state.NotImplemented: [ONNXRuntimeError] : 9 : NOT_IMPLEMENTED : Could not find an implementation for Floor(6) node with name 'Floor_4'
> ```

I'm sorry, I am still troubleshooting this.

**Comment by nijkah** [comment]
@grimoire The error raises in the `SingleRoIExtractor` which is not rewritten. I am troubleshooting to use rewriten `SingleRoIExtractor` to test mmrotate models.

**Comment by nijkah** [comment]
@grimoire @RunningLeon I may find the cause. Since [`finest_scale`](https://github.com/open-mmlab/mmdetection/blob/56e42e72cdf516bebb676e586f408b98f854d84c/mmdet/models/roi_heads/roi_extractors/single_level_roi_extractor.py#L30) is set to 56, the proposal should be larger than this.
Can I ask you to check it?

In addition to this, is it okay to insert `mmrotate` in the  [requirements/codebases.txt](https://github.com/open-mmlab/mmdeploy/blob/master/requirements/codebases.txt) to test the code in CircleCI?

**Comment by RunningLeon** [comment]
> @grimoire @RunningLeon I may find the cause. Since [`finest_scale`](https://github.com/open-mmlab/mmdetection/blob/56e42e72cdf516bebb676e586f408b98f854d84c/mmdet/models/roi_heads/roi_extractors/single_level_roi_extractor.py#L30) is set to 56, the proposal should be larger than this. Can I ask you to check it?
> 
> In addition to this, is it okay to insert `mmrotate` in the [requirements/codebases.txt](https://github.com/open-mmlab/mmdeploy/blob/master/requirements/codebases.txt) to test the code by CircleCI?

@nijkah Hi, adding `mmrotate` to `requirements/codebases.txt` is OK for me.

**Comment by nijkah** [comment]
The new requirement installation is currently canceled due to a dependency conflict between `numba` and `numpy`.
https://app.circleci.com/pipelines/github/open-mmlab/mmdeploy/1235/workflows/88534aa7-1147-47ec-855d-568b4caf027d/jobs/1983?invite=true#step-110-336

It happened in the previous [PR](https://github.com/open-mmlab/mmdeploy/pull/650)s too. [Ref](https://app.circleci.com/pipelines/github/open-mmlab/mmdeploy/1166/workflows/3e4a0b01-6754-4b28-bcd2-1d38f1fea63c/jobs/1754?invite=true#step-110-324)

Should we specify the supported `numpy` version?

**Comment by RunningLeon** [comment]
> The new requirement installation is currently canceled due to a dependency conflict between `numba` and `numpy`. https://app.circleci.com/pipelines/github/open-mmlab/mmdeploy/1235/workflows/88534aa7-1147-47ec-855d-568b4caf027d/jobs/1983?invite=true#step-110-336
> 
> It happened in the previous [PR](https://github.com/open-mmlab/mmdeploy/pull/650)s too. [CircleCI Link](https://app.circleci.com/pipelines/github/open-mmlab/mmdeploy/1166/workflows/3e4a0b01-6754-4b28-bcd2-1d38f1fea63c/jobs/1754?invite=true#step-110-324)
> 
> Should we specify the supported `numpy` version?
> 
In that case, maybe we should not add `mmrotate` in circleci. This PR would be tested locally. The numpy version conflicts in CI should be coped with carefully in another PR.

**Review by RunningLeon — APPROVED** [review]
LGTM

**Comment by nijkah** [comment]
I am still troubleshooting [the line](https://github.com/open-mmlab/mmdetection/blob/56e42e72cdf516bebb676e586f408b98f854d84c/mmdet/models/roi_heads/roi_extractors/single_level_roi_extractor.py#L53) of `SingleRoIExtractor.map_roi_levels`.

The failure reason of the test code is that the tensor is casted to Double type, although it is float type in `tests/test_codebase/test_mmdet/test_mmdet_models.py::test_cascade_roi_head`.

After changing the line as
```
        target_lvls = torch.floor(torch.log2(scale / self.finest_scale + 1e-6)).float()
```
I found it works.

Changing the test sample for proposals and casting it to float type doesn't work. 😢

**Comment by RunningLeon** [comment]
> I am still troubleshooting the test code related to [the line](https://github.com/open-mmlab/mmdetection/blob/56e42e72cdf516bebb676e586f408b98f854d84c/mmdet/models/roi_heads/roi_extractors/single_level_roi_extractor.py#L53) of `SingleRoIExtractor.map_roi_levels`. (It works well in tools/deploy.py and tools/test.py)
> 
> The failure reason of the test code is that the tensor is cast to Double type, although it is float type in `tests/test_codebase/test_mmdet/test_mmdet_models.py::test_cascade_roi_head`.
> 
> After changing the line as
> 
> ```
>         target_lvls = torch.floor(torch.log2(scale / self.finest_scale + 1e-6)).float()
> ```
> 
> I found it works.
> 
> Changing the test sample for proposals or casting it to float type doesn't work. cry So I'm still debugging it.

 `test_simple_test_of_roi_trans_roi_head` runs successfully on my side with `onnx==1.11.0` and `onnxruntime==1.8.0`

**Comment by nijkah** [comment]
Okay, I understand. The following issues will be left as future work. I'll follow these issues.

* Close performance gap between TRT and pytorch model. (Maybe due to the logic of merging patched results.)
* Add a `mmrotate` in requirements to support the unit test in CircleCI
* Check a unit test in other settings (different `onnx` and `onnxruntime` version)

**Comment by RunningLeon** [comment]
> > I am still troubleshooting the test code related to [the line](https://github.com/open-mmlab/mmdetection/blob/56e42e72cdf516bebb676e586f408b98f854d84c/mmdet/models/roi_heads/roi_extractors/single_level_roi_extractor.py#L53) of `SingleRoIExtractor.map_roi_levels`. (It works well in tools/deploy.py and tools/test.py)
> > The failure reason of the test code is that the tensor is cast to Double type, although it is float type in `tests/test_codebase/test_mmdet/test_mmdet_models.py::test_cascade_roi_head`.
> > After changing the line as
> > ```
> >         target_lvls = torch.floor(torch.log2(scale / self.finest_scale + 1e-6)).float()
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I found it works.
> > Changing the test sample for proposals or casting it to float type doesn't work. cry So I'm still debugging it.
> 
> `test_simple_test_of_roi_trans_roi_head` runs successfully on my side with `onnx==1.11.0` and `onnxruntime==1.8.0`
@nijkah 
Seems OK with me.

**Comment by nijkah** [comment]
@RunningLeon Oh, I apologize. I meant that I will follow related issues after this PR is merged.
I agree that the PR is ready to be merged.

**Review by grimoire — APPROVED** [review]
LGTM

**Pull request merged**